R/impact_method2.R

```diff
@@ -0,0 +1,442 @@
+  
+  ## cap fvps by population, and coverage by 1 - eg. AFG 2015 Measles campagin ; HTI Measles 2025
+  ## This operation makes sense - you cannot vaccinate more people than population
+  ## In addition, we need to cap coverage by 1 for method3 implementation (not happening for now)
+  i <- tab$coverage > 1. & !is.na(tab$coverage)
+  tab$coverage[i] <- 1
+  tab$fvps[i] <- tab$population[i]
+  
+  ## 8. make a local temporary table
+  ## so that we only manipulate pop-coverage-fvps once
+  DBI::dbWriteTable(con, "temporary_coverage_fvps", tab, temporary = TRUE, overwrite = TRUE)
+  message("Temporary table created - Success!")
+  
+}
+
+impact_output <- function(impact, method) {
+  ## bind output
+  impact_full <- NULL
+  impact_simplified <- NULL
+  for(i in seq_along(impact)){
+    impact_full <- rbind(impact[[i]]$impact_full, impact_full)
+    impact_simplified <- rbind(impact[[i]]$impact_simplified, impact_simplified)
+  }
+  ## order output
+  if(method == "method1") {
+    impact_full <- impact_full[order(impact_full$index, impact_full$country, impact_full$year, impact_full$age), ]
+    impact_simplified <- impact_simplified[order(impact_simplified$index, impact_simplified$impact_type, impact_simplified$country, impact_simplified$year), ]
+    
+  } else {
+    impact_full <- impact_full[order(impact_full$index, impact_full$support_type, impact_full$country, impact_full$year, impact_full$age), ]
+    impact_simplified <- impact_simplified[order(impact_simplified$index, impact_simplified$impact_type, impact_simplified$support_type, impact_simplified$country, impact_simplified$year), ]
+    
+  }
+  row.names(impact_full) <- NULL
+  row.names(impact_simplified) <- NULL
+  return(list(impact_full = impact_full, impact_simplified = impact_simplified))
+}
+
+line_up <- function(recipe, group, burden_outcomes) {
+  # The line_up function will subject to change, once impact recipes are imported into montagu
+  stopifnot(nrow(recipe) == 1L)
+  ### burden_outcome
+  burden_outcome <- unlist(strsplit(recipe$burden_outcome, ","))
+  j <- duplicated(burden_outcome)
+  if(any(j)) stop("Duplicated burden_outcomes not expected.")
+  
+  j <- match(burden_outcome, burden_outcomes$code)
+  k <- is.na(j)
+  if(any(k)) stop("Unexpected burden outcome in recipe file.")
+  burdens <- data.frame(index = recipe$index,
+                        burden_outcome_id = burden_outcomes$id[j])
+  
+  ### meta ids
+  ids <- c("responsibility_set_id", "responsibility_id", "burden_estimate_set_id", "stochastic_burden_estimate_set_id")
+  
+  j <- recipe$touchstone == group$touchstone &
+    recipe$modelling_group == group$modelling_group &
+    recipe$disease == group$disease &
+    recipe$baseline == group$scenario
+  v <- unique(group[j, ][ids])
+  if(nrow(v) == 0) stop("Impact recipe error: no known baseline burden_estimate_set.")
+  stopifnot(nrow(v) == 1L)
+  v$coeff <- 1
```

Comment:
would you mind making this `coef` (one `f`) just because this is the more usual abbreviation and I will mistype it often if it's spelt like this? :upside_down_face:

Reply:
Done.